src/arithmetic.jl

```diff
@@ -453,28 +507,53 @@ end
 # Internal multiplication functions
 for T in (:Taylor1, :TaylorN)
     # NOTE: For $T = TaylorN, `mul!` *accumulates* the result of a * b in c[k]
-    @eval @inline function mul!(c::$T{T}, a::$T{T}, b::$T{T}, k::Int) where {T<:Number}
+    @eval @inline function mul!(c::$T{T}, a::$T{T}, b::$T{T}, k::Int; scalar::NumberNotSeries=1) where {T<:Number}
```

Comment:
Eventhough I think `mul_scalar!` is better in this context (too many `mul!` methods, I find it ok to have the keyword argument.

Reply:
Thanks for this comment, I agree that it's better not to "over-overload" `mul!` so actually in the end I did add `mul_scalar!` and also the equivalent `div_scalar!`.